spec/helpers/session_timeout_warning_helper_spec.rb

```diff
@@ -20,29 +20,41 @@ def time_between_warning_and_timeout
   end
 
   describe '#timeout_refresh_url' do
-    before { expect(helper).to receive(:request).and_return(double(original_url: original_url)) }
+    before do
+      allow(helper).to receive(:request).and_return(
+        double(original_url: original_url, query_parameters: query_parameters)
+      )
+    end
+
+    let(:query_parameters) { { issuer: 'http://localhost:3000' } }
 
     context 'with no query in the request url' do
       let(:original_url) { 'http://test.host/foo/bar' }
 
-      it 'adds timeout=true params' do
-        expect(helper.timeout_refresh_url).to eq('http://test.host/foo/bar?timeout=true')
+      it 'adds timeout=true and sp_name params' do
+        expect(helper.timeout_refresh_url).to eq(
+          'http://test.host/foo/bar?issuer=http%3A%2F%2Flocalhost%3A3000&timeout=true'
+        )
       end
     end
 
     context 'with params request url' do
       let(:original_url) { 'http://test.host/foo/bar?key=value' }
 
       it 'adds timeout=true param' do
-        expect(helper.timeout_refresh_url).to eq('http://test.host/foo/bar?key=value&timeout=true')
+        expect(helper.timeout_refresh_url).to eq(
+          'http://test.host/foo/bar?issuer=http%3A%2F%2Flocalhost%3A3000&key=value&timeout=true'
+        )
       end
     end
 
-    context 'with timeout=true in the query params already' do
-      let(:original_url) { 'http://test.host/foo/bar?timeout=true' }
+    context 'with timeout=true and issuer=http://localhost:3000 in the query params already' do
+      let(:original_url) { 'http://test.host/foo/bar?timeout=true&issuer=http://localhost:3000' }
```

Comment:
I would escape the param here for consistency: `issuer=http%3A%2F%2Flocalhost%3A3000`